packages/components/addon/components/hds/dialog/header.hbs

```diff
@@ -0,0 +1,11 @@
+<div class="hds-dialog__header" ...attributes>
+  {{#if @icon}}
+    <FlightIcon class="hds-dialog__icon" @name={{@icon}} @size="24" @isInlineBlock={{false}} />
+  {{/if}}
+  <span class="hds-dialog__title" id={{@id}}>
```

Comment:
I've been pondering about this. I guess semantically it makes sense for this to be heading and as it should only be a single H1 on the page, the best candidate would be an H2, but depending on the page hierarchy it could also be an H3.

My worry with using a heading tag (regardless of the level) is that users of this component may place any content in the modal (let's say an H4 followed by a paragraph), increasing the chances of skipping headings which leads to further confusion for assistive tech users. Not setting it as part of the component, means (in my mind) that users could pass an H2 as part of the `<D.Header>` block.

I'm on the fence with this one too. @MelSumner how do you feel about this?

Reply:
I don't think we can assign a heading tag **_unless_** we have a way to know where that heading will fall into the hierarchy of the page as a whole and provide the correct heading level. 

If we're going to allow the user to nest a heading element themselves when they invoke a component, the wrapper should be a div and not a span, since a span should only contain [phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2) and a heading is not in that category.

For this specific issue, the outcomes we're after from an a11y perspective are:
- DOM order matches reading order
- Heading levels are in order (H1-H6) 

Documentation about headings and page structure: https://www.w3.org/WAI/tutorials/page-structure/headings/ 

For general information purposes, here's a video that shows how users with screen readers use headings to navigate content: https://youtu.be/QP1rzxMRap4

Reply:
Agree with the usage of a `div` element as we'll allow users to nest a heading element. Have updated accordingly!